numpy/core/tests/test_indexing.py

```diff
@@ -547,6 +547,25 @@ def test_object_assign(self):
         b[[0], :3] = [[1, (1,2), 3]]
         assert_array_equal(a, b[0])
 
+        # Check that swapping of axes works.
+        # This used to throw an error.
+        b = b.T
+        try:
+            b[:3, [0]] = [[1], [(1,2)], [3]]
+        except ValueError:
+            raise AssertionError("Assignment is possible")
```

Comment:
OK, just removed it and made the comment a bit more clear + mentioning of the gh-issue.